lib/sycamore/sycamore/transforms/assign_doc_properties.py

```diff
@@ -0,0 +1,52 @@
+from sycamore.data import Document
+from sycamore.plan_nodes import Node, SingleThreadUser, NonGPUUser
+from sycamore.transforms.map import Map
+from sycamore.utils.time_trace import timetrace
+import json
+from typing import List, Dict
+
+
+class AssignDocProperties(SingleThreadUser, NonGPUUser, Map):
+    """
+    The AssignDocProperties transform is used to copy properties from specific element stored
+    within a JSON string to the parent document. This allows for the consolidation of key
+    attributes at the document level.
+
+    Args:
+        child: The source node or component that provides the hierarchical documents to be exploded.
+        resource_args: Additional resource-related arguments that can be passed to the explosion operation.
+
+    Example:
+        .. code-block:: python
+
+            source_node = ...  # Define a source node or component that provides hierarchical documents.
+            property_transform = AssignDocProperties(child=source_node, list=["table"])
+            property_dataset = property_transform.execute()
+    """
+
+    def __init__(self, child: Node, parameters: List[str], **resource_args):
+        super().__init__(child, f=AssignDocProperties.assign_doc_properties, args=parameters, **resource_args)
+
+    @staticmethod
+    def _parse_json(s: str) -> Dict:
+        s = s[next(idx for idx, c in enumerate(s) if c in "{[") :]
```

Comment:
this might not be needed now, in the repurposed [assign_doc_properties.py](https://github.com/aryn-ai/sycamore/pull/645/files/c0561a0edcf720fa7d985a1f42e7a571520f9f13#diff-4df39ca3f1bfd8b62fc663fac55bb7bb84473d54dc2a2165ed85277387e36c2f) as in the newer implementation we are copying properties directly from child to parent.